src/actions/common.js

```diff
@@ -185,10 +185,14 @@ export const loadInitialData = clientId => (dispatch, getState) => {
       return
     }
 
-    // In embedded chat conf.channelId is defined.
-    const channelId = conf.channelId || findLastUsedChannel(channelsSelector(getState())).id
-
-    dispatch(setChannel(channelId))
+    const {route} = appSelector(getState())
+    if (route.params.channelId) {
```

Comment:
Shouldn't this be based on if conf.channelId is defined use it? Otherwise someone using an embedded chat could use the channelId param in the route to load another channel?!?

Reply:
We are not using a router for the embedded client - https://github.com/ubergrape/grape-web-client/blob/master/src/app/AppEmbedded.js#L33. We wrote own component to emulate router:D https://github.com/ubergrape/grape-web-client/blob/88213ca4bfa4aa96bd5dc414eca078f11fe49db3/src/containers/router/FakeRouter.js

Reply:
Still I would suggest to flip it, because without knowing that there is a fake router this is confusing and every future engineer might struggle with this.

```
// channelId in conf is only set in the embedded chat
if (conf.channelId) {
  …
} else if (route.params.channelId) {
  dispatch(setChannel(route.params.channelId))
} else {
   // throw exception here
}
```

It's bit more verbose, but very explicit that in the embedded chat you can't change the route.

Reply:
@nikgraf sorry, didn't get what should I do inside first if statement? I mean this one: `if (conf.channelId)`